src/test/regress/expected/shard_rebalancer.out

```diff
@@ -2712,6 +2713,105 @@ SELECT sh.logicalrelid, pl.nodeport
  single_shard_colocation_2b |    57637
 (5 rows)
 
+DROP TABLE single_shard_colocation_1a, single_shard_colocation_1b, single_shard_colocation_1c, single_shard_colocation_2a, single_shard_colocation_2b CASCADE;
+-- test the same with coordinator shouldhaveshards = false and shard_count = 2
+-- so that the shard allowed node count would be 2 when rebalancing
+-- for such cases, we only count the nodes that are allowed for shard placements
+UPDATE pg_dist_node SET shouldhaveshards=false WHERE nodeport = :master_port;
+SELECT 1 FROM citus_remove_node('localhost', :worker_2_port);
+ ?column?
+---------------------------------------------------------------------
+        1
+(1 row)
+
+create table single_shard_colocation_1a (a int primary key);
+create table single_shard_colocation_1b (a int primary key);
+create table single_shard_colocation_1c (a int primary key);
+SET citus.shard_replication_factor = 1;
+select create_distributed_table('single_shard_colocation_1a','a', colocate_with => 'none', shard_count => 2);
```

Comment:
the table names are pretty confusing since they have two shards:
```suggestion
select create_distributed_table('two_shard_colocation_1a','a', colocate_with => 'none', shard_count => 2);
```